R/mrgsims.R

```diff
@@ -241,10 +241,23 @@ as_data_frame.mrgsims <- function(.data_,...) {
   as_tibble(as.data.frame(.data_),...)
 }
 
-##' @rdname mrgsims_dplyr
-##' @export
-as_tibble.mrgsims <- function(.data_,...) {
-  as_tibble(as.data.frame(.data_),...)  
+#' @param .rows passed to [tibble::as_tibble()]
+#' @param .name_repair passed to [tibble::as_tibble()]
+#' @param rownames passed to [tibble::as_tibble()]
+#' @rdname mrgsims_dplyr
+#' @export
+as_tibble.mrgsims <- function(x, 
+                              .rows = NULL, 
+                              .name_repair = c("check_unique", "unique", 
+                                               "universal", "minimal"), 
+                              rownames = NULL, ...) {
+  as_tibble(
+    as.data.frame(x), 
+    .rows = .rows, 
+    .name_repair = .name_repair, 
+    rownames = rownames, 
+    ...
+  )  
 }
```

Comment:
Was it just unhappy about the name of the first parameter?  I can get the warning to go away on my end witth just `(x, ...)` as the signature.

Reply:
Thanks for testing that out. I thought methods had to have all the arguments of the generic. 

https://rstudio.github.io/r-manuals/r-exts/Generic-functions-and-methods.html

I could have sworn I ran into this before ...

Reply:
@kyleam I went with the simpler fix as you suggested. I don't know what the status is of the `as_data_frame()` generic ... so I changed that argument too. I don't care if that stays or goes at this point so I just changed it to try to stay out of trouble. 


```r
> dplyr::as_data_frame
function(x, ...) {
  deprecate_warn("2.0.0", "as_data_frame()", "as_tibble()",
    details = "The signature and semantics have changed, see `?as_tibble`."
  )

  as_tibble(x, ...)
}
```

I sent this to R-devel to make sure it passed there. Will update when that comes back.

Reply:
> I thought methods had to have all the arguments of the generic.
>
> https://rstudio.github.io/r-manuals/r-exts/Generic-functions-and-methods.html

My understanding is that, although a method needs to accept all the arguments of the generic, it's okay for a method to fold arguments into `...`.  In the link above, using `...` in this scenario is referenced a bit in the `predict.lm` problem that's mentioned: "as `predict.lm` had neither a `dispersion` nor a `...` argument, `NextMethod` could no longer be used".

Note that some of tibble's own methods use `(x, ...)`:

```
tibble $ git describe HEAD
v3.2.0.9000

tibble $ git grep '^as_tibble\.' 'R/*.R'
R/as_tibble.R:as_tibble.data.frame <- function(x, validate = NULL, ...,
R/as_tibble.R:as_tibble.list <- function(x, validate = NULL, ..., .rows = NULL,
R/as_tibble.R:as_tibble.matrix <- function(x, ..., validate = NULL, .name_repair = NULL) {
R/as_tibble.R:as_tibble.poly <- function(x, ...) {
R/as_tibble.R:as_tibble.ts <- function(x, ..., .name_repair = "minimal") {
R/as_tibble.R:as_tibble.table <- function(x, `_n` = "n", ..., n = `_n`, .name_repair = "check_unique") {
R/as_tibble.R:as_tibble.NULL <- function(x, ...) {
R/as_tibble.R:as_tibble.default <- function(x, ...) {
```

Reply:
Great. This really simplifies things. I'll merge this and then send it to CRAN; the winbuilder (failed on 1.0.8) came back fine now.